zebra-network/src/meta_addr.rs

```diff
@@ -103,6 +104,42 @@ pub struct MetaAddr {
     /// records, older peer versions, or buggy or malicious peers.
     pub services: PeerServices,
 
+    /// The last time we interacted with this peer.
+    ///
+    /// See `get_last_seen` for details.
+    last_seen: DateTime<Utc>,
+
+    /// The outcome of our most recent communication attempt with this peer.
+    pub last_connection_state: PeerAddrState,
+}
+
+impl MetaAddr {
+    /// Create a new `MetaAddr` from the deserialized fields in an `Addr`
+    /// message.
+    pub fn new_gossiped(
+        addr: &SocketAddr,
+        services: &PeerServices,
+        last_seen: &DateTime<Utc>,
+    ) -> MetaAddr {
+        MetaAddr {
+            addr: *addr,
+            services: *services,
+            last_seen: *last_seen,
+            // the state is Zebra-specific, it isn't part of the Zcash network protocol
+            last_connection_state: NeverAttempted,
```

Comment:
Use explicit `NeverAttempted` rather than `PeerAddrState::default()` for readability.